### ISSUE TITLE
Wrongly assumed indentation with match and arrays

### ISSUE THREAD

**Issue by iquito** [squizlabs/PHP_CodeSniffer#3287]
Using PSR-12 with current `dev-master` the PHP8 `match` clause and arrays have a "wrong" indentation interaction:

```php
class TestClass
{
    public function dada(string $type): array
    {
        return match ($type) {
            'referrers' => [
                'referrersNumber' => 'DESC',
                'visitorsNumber' => 'DESC',
                'viewsNumber' => 'DESC',
            ],
            'orders' => [
                'ordersNumber' => 'DESC',
                'visitorsNumber' => 'DESC',
                'viewsNumber' => 'DESC',
                'referrersNumber' => 'DESC',
            ],
            default => [
                'visitorsNumber' => 'DESC',
                'viewsNumber' => 'DESC',
                'referrersNumber' => 'DESC',
            ],
        };
    }
}
```

Reports the following errors:

```
 18 | ERROR | [x] Line indented incorrectly; expected at least 16 spaces, found 12
 19 | ERROR | [x] Line indented incorrectly; expected at least 16 spaces, found 12
 23 | ERROR | [x] Line indented incorrectly; expected at least 16 spaces, found 12
```

The lines reported are the one before the default clause, the line with the default clause, and the ending array of the default clause. There errors always seem to start at the end of the second array, with or without a default clause. When using other types instead of arrays no errors occur.

**Comment by gsherwood** [comment]
Thanks a lot for testing master and reporting this. This was caused by match expression support in findStartofStatement and findEndOfStatement being incorrect. Once fixing those, the sniff reports indentation correctly.
